sentry_sdk/integrations/aws_lambda.py

```diff
@@ -116,6 +116,13 @@ def sentry_handler(event, context, *args, **kwargs):
                     )
                     hub.capture_event(event, hint=hint)
                     reraise(*exc_info)
+                finally:
+                    if (
+                        integration.timeout_warning
+                        and configured_time > TIMEOUT_WARNING_BUFFER
+                    ):
+                        timeout_thread.stop_thread = True
+                        timeout_thread.join()
```

Comment:
This could potentially block the main thread of the lambda handler, which is undesirable.